### ISSUE TITLE
read_delim: unable to parse thorn delimiter - þ

### ISSUE THREAD

**Issue by kadekillary** [tidyverse/readr#820]
I've tried a variety of things but read_delim seems unable to parse this particularly nasty character.

Character encoding is latin1 (ISO-8859-1) and two bytes. `\xfe` is unicode representation.

All of the following fail:

`invalid multibyte string in col_name`
read_delim('thorn_test.txt', 'þ')

`parses all into one column`
read_delim('thorn_test.txt', '\xfe')

`correct columns - appends (¾) char in every column/value`
read_delim('thorn_test.txt', 'þ', locale=locale(encoding='latin1'))
read_delim('thorn_test.txt', 'þ', locale=locale(encoding='ISO-8859-1'))

`one column - appends  (Ã¾) in every column/value`
read_delim('thorn_test.txt', '\xfe', locale=locale(encoding='latin1'))

Conversely, in Python using Pandas the following works:

`pandas.read_csv('~/thorn_test.txt', delim='\xfe', engine='python')`

However, it doesn't work if the default C engine is used. Therefore, I'm thinking it has to do with how C/C++ handles this character.

**Comment by batpigandme** [comment]
Could you please turn this into a self-contained [reprex](http://reprex.tidyverse.org/) (short for minimal **repr**oducible **ex**ample)? It will help us help you if we can be sure we're all working with/looking at the same stuff.

If you've never heard of a reprex before, you might want to start by reading the tidyverse.org [help page](https://www.tidyverse.org/help/). The [reprex dos and don'ts](http://reprex.tidyverse.org/articles/reprex-dos-and-donts.html) are also useful.

**Comment by beigebrucewayne** [comment]
Data used to generate examples: [here](https://gist.github.com/beigebrucewayne/1ccd00ac1e14ad0cdc9a28a87f809af6)

```r
library(readr)
# [1] "readr"     "nvimcom"   "stats"     "graphics"  "grDevices" "utils"    
# [7] "datasets"  "methods"   "base"     

read_delim('~/Desktop/thorn_test.txt', 'þ')
# Error in make.names(x) : invalid multibyte string at '<be>las<74>_name'

read_delim('~/Desktop/thorn_test.txt', '\xfe')
# # A tibble: 4 x 1
#   first_nameþlast_nameþsalary
#   <chr>                      
# 1 jamesþbondþ100000000000    
# 2 ronaldþmcdonaldþ1010202    
# 3 lebronþjamesþ203030303     
# 4 helmutþnewtonþ2374843      

read_delim('~/Desktop/thorn_test.txt', 'þ', locale=locale(encoding='latin1'))
# # A tibble: 4 x 3
#   first_name `¾last_name` `¾salary`    
#   <chr>      <chr>        <chr>        
# 1 james      ¾bond        ¾100000000000
# 2 ronald     ¾mcdonald    ¾1010202     
# 3 lebron     ¾james       ¾203030303   
# 4 helmut     ¾newton      ¾2374843     

read_delim('~/Desktop/thorn_test.txt', 'þ', locale=locale(encoding='ISO-8859-1'))
# # A tibble: 4 x 3
#   first_name `¾last_name` `¾salary`    
#   <chr>      <chr>        <chr>        
# 1 james      ¾bond        ¾100000000000
# 2 ronald     ¾mcdonald    ¾1010202     
# 3 lebron     ¾james       ¾203030303   
# 4 helmut     ¾newton      ¾2374843     

read_delim('~/Desktop/thorn_test.txt', '\xfe', locale=locale(encoding='latin1'))
# # A tibble: 4 x 1
#   `first_nameÃ¾last_nameÃ¾salary`
#   <chr>                          
# 1 jamesÃ¾bondÃ¾100000000000      
# 2 ronaldÃ¾mcdonaldÃ¾1010202      
# 3 lebronÃ¾jamesÃ¾203030303       
# 4 helmutÃ¾newtonÃ¾2374843        

read_delim('~/Desktop/thorn_test.txt', '\xfe', locale=locale(encoding='ISO-8859-1'))
# # A tibble: 4 x 1
#   `first_nameÃ¾last_nameÃ¾salary`
#   <chr>                          
# 1 jamesÃ¾bondÃ¾100000000000      
# 2 ronaldÃ¾mcdonaldÃ¾1010202      
# 3 lebronÃ¾jamesÃ¾203030303       
# 4 helmutÃ¾newtonÃ¾2374843        
```

**Comment by jimhester** [comment]
readr assumes only single byte delimiters. You will have to convert these multi-byte delimiters to a single byte delimiter such as `,` or `\t` in a pre-processing step before using readr.

**Comment by ForwardDataWorks** [comment]
@kadekillary  I found it easiest to import as you described in your attempt No.3 and then deal with the "3/4" after the fact.

packages needed: tidyverse, janitor

Use janitor::clean_names to avoid getting the "3/4" in your column names.


```
> df <- clean_names(read_delim("file_name.txt", 
    delim = "\U00fe", 
    col_names = T, 
    locale = locale(encoding = "Latin1")
)
) %>%
>     mutate_all(~str_replace_all(., "\U00BE", "")
```

where 00BE is the unicode for "3/4"

**Comment by jimhester** [comment]
This is fixed as part of #1172

---
author: jhester
date: 2021-04-29
output: "reprex::reprex\\_document"
title: sane-eyas\_reprex.R
---

``` r
readr::read_delim(I("1þ2þ3\naþbþc\n"), delim = "þ")
#> Rows: 1 Columns: 3
#> ── Column specification ────────────────────────────────────────────────────────
#> Delimiter: "þ"
#> chr (3): 1, 2, 3
#> 
#> ℹ Use `spec()` to retrieve the full column specification for this data.
#> ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#> # A tibble: 1 x 3
#>   `1`   `2`   `3`  
#>   <chr> <chr> <chr>
#> 1 a     b     c
```

<sup>Created on 2021-04-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
